hashmap_design.py

```diff
@@ -0,0 +1,123 @@
+# Approach: Initially tried with using a large array defined as per constraint ([0, 1000000])
+# If the key is not found we return -1 hence intialized the huge sized array with -1
+# Time complexity - O(1) for get, put,remove
+# Space complexity - O(N) where N is the size of the array
+# Not an ideal solution as defining a huge array adds to space constraint 
+# Ran with 365 ms and 39 Mb space
+class MyHashMap:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.arr = [-1] * 1000000
+        
+
+    def put(self, key: int, value: int) -> None:
+        """
+        value will always be non-negative.
+        """
+        self.arr[key] = value
+        
+
+    def get(self, key: int) -> int:
+        """
+        Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
+        """
+        
+        return self.arr[key]
+        
+
+    def remove(self, key: int) -> None:
+        """
+        Removes the mapping of the specified value key if this map contains a mapping for the key
+        """
+        self.arr[key] = -1
+        
+
+
+# Your MyHashMap object will be instantiated and called as such:
+# obj = MyHashMap()
+# obj.put(key,value)
+# param_2 = obj.get(key)
+# obj.remove(key)
+
+#####
+
+# Approach 2: Declared an array for holding the data with the help of a linked list
+# Used a basic hashing by performing modulo operation on the key wrt to the size declared with a linked list approach
+# Not sure how to calculate Time complexity
```

Comment:
The worst case Time complexity will be linear. If assumed that all buckets hold equal number of elements, the Asymptotic Time here will be O(n/m), otherwise in the worst case if all elements are in a single bucket then O(n).